rbd.c

```diff
@@ -260,22 +260,29 @@ static void tcmu_rbd_detect_device_class(struct tcmu_device *dev)
 	}
 
 	ret = rados_mon_command(state->cluster, (const char **)mon_cmd_bufs, 1,
-				"", 0, &mon_buf, &mon_buf_len,
+				"", 0, &temp_mon_buf, &temp_mon_buf_len,
 				&mon_status_buf, &mon_status_buf_len);
 	free(mon_cmd_bufs[0]);
-	if (ret < 0 || !mon_buf) {
+	if (ret < 0 || !temp_mon_buf) {
 		tcmu_dev_warn(dev, "Could not retrieve pool crush rule (Err %d)\n",
 			      ret);
 		return;
 	}
 	rados_buffer_free(mon_status_buf);
 
 	/* expected JSON output: "{..."crush_rule":"<rule name>"...}" */
-	mon_buf[mon_buf_len] = '\0';
+	mon_buf_len = temp_mon_buf_len + 1;
+	if(!(mon_buf = (char *)malloc(mon_buf_len))) {
```

Comment:
Hey letterwuyu,

Very nice find!

Just use the standard coding style in the rest of the  file. No need to cast from void to char and just do the alloc and test on 2 lines like:

mon_buf = malloc(mon_buf_len);
if (!mon_buf) {

Also, in this memory allocate failure path, do we also need a:

rados_buffer_free(temp_mon_buf);

?